src/lib/encoder.ts

```diff
@@ -80,6 +81,55 @@ const decodeDataSourceWithHash = (value: string): URLDataWithHash => {
   return { hashFunction: hashFunction.name, hash: dataHash, url: dataSource };
 };
 
+const encodeCompactBytesArray = (values: string[]): string => {
+  let compactBytesArray = '0x';
+
+  for (let i = 0; i < values.length; i++) {
+    if (values[i] === '')
+      throw new Error(
+        `Couldn't encode, value at index ${i} is '', expected '0x'`,
```

Comment:
is "expected 0x" really true?

Do you mean "Expacted an hexadecimal string with 0x prefix instead?

Reply:
Yeah, if someone passes '', it is not valid hex value and `isHex('')` returns `true`. When encoding a '' value it has unexpected behaviour and if you want to encode an empty bytes you need to pass it as '0x' and not ''